src/main/java/org/sonarlint/intellij/actions/AbstractSonarAction.java

```diff
@@ -21,40 +21,43 @@
 
 import com.intellij.openapi.actionSystem.AnAction;
 import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAware;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.VirtualFile;
 import java.util.stream.Stream;
 import javax.swing.Icon;
 import org.jetbrains.annotations.Nullable;
 import org.sonarlint.intellij.analysis.AnalysisStatus;
 
-public abstract class AbstractSonarAction extends AnAction {
+public abstract class AbstractSonarAction extends AnAction implements DumbAware {
```

Comment:
Have you anticipated the impacts of implementing `DumbAware`? Are we sure all or actions can work properly in Dumb mode?

Reply:
I tried the actions that are available with Shift-Shift when the project is in dumb mode. For all other actions they are available through the tool window and during indexing it is not shown. So I think it should be safe